test/e2e-tests.sh

```diff
@@ -172,4 +178,13 @@ fi
 go_test_e2e ./test/e2e/... || failed=1
 (( failed )) && fail_test
 
+# Re-run the PipelineRun tests with "embedded-status" set to "minimal"
+header "Running Go e2e tests with Pipeline embedded-status set to minimal"
+
+set_minimal_embedded_status
```

Comment:
Should we keep testing the old behaviour too? Not sure how though, I guess that would mean an extra e2e job?

Reply:
We are still testing that - we're running the full set of tests with the default (`full`), and then updating the flag to `minimal` and rerunning the relevant tests.

Reply:
I decided it wasn't worth adding an additional job given how few tests actually change behavior depending on the `embedded-status` value - it was markedly simpler and faster to get what we needed by just doing it this way. =)